### PR TITLE
implemented checking rooms available @viewListings

### PR DESCRIPTION
**Pull request by dylanjian0**

### CODE REVIEW

File: src/components/viewListings.js [hunk 1]
```diff
@@ -71,8 +71,8 @@ export default function ViewListings(){
                                 <p className='text-xs font-bold text-green-500'> {listing.rooms - listing.joinedListing.length} / {listing.rooms} Rooms Open </p>
                                 <p className='text-xs'> {listing.address} </p>
                                 <p className='text-xs'> {listing.city}, {listing.state}, {listing.zipCode}</p>
-                                { listing.joinedListing.includes(user.nickname)
-                                ? <p className='text-gray-400 text-xs font-bold text-center'> Already Joined </p>
+                                { listing.joinedListing.includes(user.nickname) || listing.rooms - listing.joinedListing.length <= 0
```
Comment (joonhoswe) [inline src/components/viewListings.js:74]: Listing would/should disappear once it hits 0 rooms left, also making it display 2 statuses such as already joined / no room at the same time is not ideal

Comment (dylanjian0) [reply src/components/viewListings.js:74]: true, let's talk about that later

Comment (dylanjian0) [reply src/components/viewListings.js:74]: on the call

File: src/components/viewListings.js [hunk 2]
```diff
@@ -61,22 +79,30 @@ export default function ViewListings(){
...
+                                <p className={`text-xs font-bold ${listing.rooms-listing.joinedListing.length === 0 ? 'text-red-500' : 'text-green-500'}`}>
+                                  {listing.rooms - listing.joinedListing.length} / {listing.rooms} Rooms Open 
+                                </p>
                                 <p className='text-xs'> {listing.address} </p>
                                 <p className='text-xs'> {listing.city}, {listing.state}, {listing.zipCode}</p>
-                                { listing.joinedListing.includes(user.nickname)
-                                ? <p className='text-gray-400 text-xs font-bold text-center'> Already Joined </p>
-                                : <button onClick={()=> handleJoin(listing.id, user.nickname)} className='text-red-500 text-xs font-bold transition ease-in-out duration-300 hover:text-gray-400'> Join </button>
+                                { listing.joinedListing.includes(user.nickname) || listing.rooms - listing.joinedListing.length <= 0 ?
```
Comment (joonhoswe) [inline src/components/viewListings.js:95]: checking if either those is true first and then checking again to see which is true is a bit redundant and opens room for error, you can simply check: if the user is joined ..., else if rooms left == 0...

File: src/components/viewListings.js [hunk 3]
```diff
@@ -61,22 +79,30 @@ export default function ViewListings(){
...
-                                { listing.joinedListing.includes(user.nickname)
-                                ? <p className='text-gray-400 text-xs font-bold text-center'> Already Joined </p>
-                                : <button onClick={()=> handleJoin(listing.id, user.nickname)} className='text-red-500 text-xs font-bold transition ease-in-out duration-300 hover:text-gray-400'> Join </button>
+                                { listing.joinedListing.includes(user.nickname) || listing.rooms - listing.joinedListing.length <= 0 ? 
+                                    listing.joinedListing.includes(user.nickname) ?
+                                    <>
+                                      <p className='text-gray-400 text-xs font-bold text-center'> Already Joined</p>
+                                      <button onClick={()=> handleLeave(listing.id, user.nickname)} className='text-red-500 text-xs font-bold transition ease-in-out duration-300 hover:text-gray-400'> Leave </button>
+                                    </> : 
+                                      <p className='text-gray-400 text-xs font-bold text-center'>Listing Full</p>
```
Comment (joonhoswe) [inline src/components/viewListings.js:101]: this is viewListing, so assuming your conditional rendering of this element works, the user shouldn't ever see "listing full"

File: src/components/viewListings.js (hunk 2 above)
Comment (dylanjian0) [reply src/components/viewListings.js:95]: fixed

File: src/components/viewListings.js (hunk 3 above)
Comment (dylanjian0) [reply src/components/viewListings.js:101]: Yup true, this code was written before I implemented hiding a listing if it was unavailable.

**Pull request merged**